tests/scripts/ci_helpers.sh

```diff
@@ -242,13 +249,76 @@ function seed_lxd_profile() {
     lxc network list
 }
 
+function prepare_environment() {
+  set -ux
+  date
+  sudo snap install juju
+  mkdir -p ~/.local/share/juju
+  
+  declare -i i=0
+  while [[ $i -lt 20 ]]; do
+      seed_lxd_profile ./tests/scripts/assets/lxd-preseed.yaml
+      if [[ $? -eq 0 ]]; then
+        echo "LXD profile seeding successful."
+        break
+      fi
+      echo "failed $i attempt, retrying in 5 seconds";
+      sleep 5s
+      i=$((i + 1));
+  done
+
+  if [[ $i -eq 20 ]]; then
+      echo "Timeout reached, failed to seed lxd profile."
+      exit -1
+  fi
+
+  juju bootstrap localhost lxd
+
+  chmod 600 ./tests/scripts/assets/id_rsa*
+}
+
 function wait_for_microceph_bootstrap() {
     # wait for install hook to trigger on one unit.
     juju wait-for unit microceph/0 --query='workload-message=="(install) (bootstrap) Service not bootstrapped"' --timeout=20m
     # Now wait for the model to settle.
     juju wait-for application microceph --query='name=="microceph" && (status=="active" || status=="idle")' --timeout=10m
 }
 
+function prepare_3_vms() {
+  set -ux
+
+  lxc launch --vm ubuntu:24.04 node01 -c limits.memory=8GB -c limits.cpu=4 -d root,size=32GB
+  lxc launch --vm ubuntu:24.04 node02 -c limits.memory=8GB -c limits.cpu=4 -d root,size=32GB
+  lxc launch --vm ubuntu:24.04 node03 -c limits.memory=8GB -c limits.cpu=4 -d root,size=32GB
```

Comment:
Could we make do with smaller root size disks? Asking b/c the xlarge runners have 100G of total disk, and 3x32G for VMs doesn't leave a lot of buffer. Otoh, we're not creating huge OSDs so we could fit them onto smaller potentially